src/com/amazon/ion/system/IonReaderBuilder.java

```diff
@@ -157,6 +168,110 @@ private IonCatalog validateCatalog()
         return catalog != null ? catalog : new SimpleCatalog();
     }
 
+    /**
+     * <p>
+     * Determines whether the IonReader will allow incremental reading of binary Ion data. When enabled, if
+     * {@link IonReader#next()} returns {@code null} at the top-level, it indicates that there is not enough data
+     * in the stream to complete a top-level value. The user may wait for more data to become available in the stream
+     * and call {@link IonReader#next()} again to continue reading. Unlike the non-incremental reader, the incremental
+     * reader will never throw an exception due to unexpected EOF during {@code next()}. If, however,
+     * {@link IonReader#close()} is called when an incomplete value is buffered, an {@link IonException} will be raised.
+     * </p>
+     * <p>
+     * There is no incremental text IonReader, so for text data a non-incremental IonReader will always be returned
+     * regardless of the value of this option.
+     * </p>
+     * </p>
+     * When this option is enabled, auto-detection of GZIP data is not supported; the byte array or InputStream
+     * implementation provided to {@link #build(byte[])} or {@link #build(InputStream)} must return uncompressed bytes.
+     * This can be achieved by wrapping the data in a GZIPInputStream and passing it to {@link #build(InputStream)}.
```

Comment:
Is there a reason we can't do this wrapping on the user's behalf?

Reply:
Technically, no. I would just prefer not to add more new code that does this. It's always felt arbitrary (why not Zstd? LZ4? etc.), and it's caused [problems](https://github.com/amzn/ion-java/blob/master/src/com/amazon/ion/IonSystem.java#L272) with other APIs in the past.